tests/pytest.ini

```diff
@@ -2,4 +2,5 @@
 markers =
     mgpu: Mark a test that requires multiple GPUs to run.
     ci: Mark a test that runs only on CI.
-    gtest: Mark a test that requires C++ Google Test executable.
\ No newline at end of file
+    gtest: Mark a test that requires C++ Google Test executable.
+    no_rmm_pool_setup: Mark a test to skip the setup_rmm_pool() fixture.
```

Comment:
Please add a note when should it apply.  From I have seen it should be applied when:
- dask is used.
- demo that doesn't use rmm.

Correct me if I'm wrong. ;-)

Reply:
Mainly this mark is to avoid allocating RMM pool twice. If a Dask cluster sets up an RMM pool, then we do not want to set up another pool again (that will cause OOM).

Reply:
Em ... I need to be extra careful when writing tests ...

Reply:
@trivialfis Can we use a fixture for Dask clusters? Then the RMM fixture can reliably detect whether a Dask cluster is being used.

Reply:
Yes, there's document in distributed library's developer guide.

Reply:
@hcho3 See `tests/python/test_with_dask.py` for an example.  I'm not sure about cuda cluster though.

Reply:
Also https://github.com/rapidsai/cuml/blob/branch-0.15/python/cuml/test/dask/conftest.py ?

Reply:
Done. Now `setup_rmm_pool()` fixture will ignore any test that uses `local_cuda_cluster` fixture.

Reply:
Actually, I ended up fixing the issue by reducing the pool size. Duplicated pools get resolved eventually as Python's garbage collection removes the old pool. The default size (1/2 GPU memory) is too big. Setting the pool to be < 4GB does the trick.